awswrangler/lakeformation/_utils.py

```diff
@@ -1,18 +1,134 @@
+) -> str:
+    partition_predicates: List[str] = []
+    for col, val in zip(partition_cols, partitions_values):
+        if partitions_types[col].startswith(("tinyint", "smallint", "int", "bigint", "float", "double", "decimal")):
+            partition_predicates.append(f"{col}={str(val)}")
+        else:
+            partition_predicates.append(f"{col}='{str(val)}'")
+    return " AND ".join(partition_predicates)
+
+
+def _build_table_objects(
+    paths: List[str],
+    partitions_values: Dict[str, List[str]],
+    use_threads: bool,
+    boto3_session: Optional[boto3.Session],
+) -> List[Dict[str, Any]]:
+    table_objects: List[Dict[str, Any]] = []
+    paths_desc: Dict[str, Dict[str, Any]] = describe_objects(
+        path=paths, use_threads=use_threads, boto3_session=boto3_session
+    )
+    for path, path_desc in paths_desc.items():
+        table_object: Dict[str, Any] = {
+            "Uri": path,
+            "ETag": path_desc["ETag"],
+            "Size": path_desc["ContentLength"],
+        }
+        if partitions_values:
+            table_object["PartitionValues"] = partitions_values[f"{path.rsplit('/', 1)[0].rstrip('/')}/"]
+        table_objects.append(table_object)
+    return table_objects
+
+
+def _get_table_objects(
+    catalog_id: Optional[str],
+    database: str,
+    table: str,
+    transaction_id: str,
+    boto3_session: Optional[boto3.Session],
+    partition_cols: Optional[List[str]] = None,
+    partitions_types: Optional[Dict[str, str]] = None,
+    partitions_values: Optional[List[str]] = None,
+) -> List[Dict[str, Any]]:
+    """Get Governed Table Objects from Lake Formation Engine."""
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+
+    scan_kwargs: Dict[str, Union[str, int]] = _catalog_id(
+        catalog_id=catalog_id,
+        **{
+            "TransactionId": transaction_id,
+            "DatabaseName": database,
+            "TableName": table,
+            "MaxResults": 100,
+        },
+    )
+    if partition_cols and partitions_types and partitions_values:
+        scan_kwargs["PartitionPredicate"] = _build_partition_predicate(
+            partition_cols=partition_cols, partitions_types=partitions_types, partitions_values=partitions_values
+        )
+
+    next_token: str = "init_token"  # Dummy token
+    table_objects: List[Dict[str, Any]] = []
+    while next_token:
+        response = client_lakeformation.get_table_objects(**scan_kwargs)
+        for objects in response["Objects"]:
+            for table_object in objects["Objects"]:
+                if objects["PartitionValues"]:
+                    table_object["PartitionValues"] = objects["PartitionValues"]
+                table_objects.append(table_object)
+        next_token = response.get("NextToken", None)
+        scan_kwargs["NextToken"] = next_token
+    return table_objects
```

Comment:
Returns objects currently registered in the Lake Formation engine for a given Glue Governed table and a specific `partition_predicate`